src/pydargs/__init__.py

```diff
@@ -35,21 +35,89 @@ class DataClassProtocol(Protocol):
 Dataclass = TypeVar("Dataclass", bound=DataClassProtocol)
 
 
-def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, **kwargs: Any) -> Dataclass:
+def parse(
+    tp: Type[Dataclass], args: Optional[list[str]] = None, load_from_config: bool = False, **kwargs: Any
+) -> Dataclass:
     """Instantiate an object of the provided dataclass type from command line arguments.
 
     Args:
         tp: Type of the object to instantiate. This is expected to be a dataclass.
         args: Optional list of arguments. Defaults to sys.argv.
+        load_from_config: If True, add a --config-file argument to load defaults from a JSON- or YAML-formatted file.
         **kwargs: Keyword arguments passed to the ArgumentParser object.
 
     Returns:
         An instance of tp.
     """
-    namespace = _create_parser(tp, **kwargs).parse_args(args)
+    namespace = _create_parser(tp, load_from_config=load_from_config, **kwargs).parse_args(args)
+    if load_from_config:
+        _add_defaults_from_file(namespace)
```

Comment:
Does this have to happen in-place? I think this would be clearer:
```suggestion
    if load_from_config:
        namespace = _add_defaults_from_file(namespace)
```

Reply:
It doesn't _have to_, in fact I spent some time this morning refactoring so it copies the namespace... but I think this fits the current situation better because:
- It doesn't require copying the namespace twice, and
- the namespace is already modified in-place by `_create_object`.

What do you think?